### ISSUE TITLE
Use proper typings

### ISSUE THREAD

**Issue by ghost** [SkylixGH/LuxJS#10]
Convert such to such

**FROM EX**
```
function test(settings: Settings) {}
```

**TO EX**
```
function test(settings: Partial<Settings>) {}
```

And make all `Settings` interfaces not contain any optional parameters

**Comment by DarkMalwre** [comment]
ye use proper typings sir

**Comment by ghost** [comment]
xd

**Comment by DarkMalwre** [comment]
tbh I have no clue what you're talking about lmao

**Comment by ghost** [comment]
This has been implemented and is available and confirmed to be working in v1.0.0-dev.2
